pkg/client/clientset/versioned/typed/rook/v1alpha1/cluster.go

```diff
@@ -0,0 +1,155 @@
+/*
+Copyright 2017 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	v1alpha1 "github.com/rook/rook/pkg/apis/rook.io/v1alpha1"
+	scheme "github.com/rook/rook/pkg/client/clientset/versioned/scheme"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	types "k8s.io/apimachinery/pkg/types"
+	watch "k8s.io/apimachinery/pkg/watch"
+	rest "k8s.io/client-go/rest"
+)
+
+// ClustersGetter has a method to return a ClusterInterface.
+// A group's client should implement this interface.
+type ClustersGetter interface {
+	Clusters(namespace string) ClusterInterface
+}
+
+// ClusterInterface has methods to work with Cluster resources.
+type ClusterInterface interface {
+	Create(*v1alpha1.Cluster) (*v1alpha1.Cluster, error)
+	Update(*v1alpha1.Cluster) (*v1alpha1.Cluster, error)
+	Delete(name string, options *v1.DeleteOptions) error
+	DeleteCollection(options *v1.DeleteOptions, listOptions v1.ListOptions) error
+	Get(name string, options v1.GetOptions) (*v1alpha1.Cluster, error)
+	List(opts v1.ListOptions) (*v1alpha1.ClusterList, error)
+	Watch(opts v1.ListOptions) (watch.Interface, error)
+	Patch(name string, pt types.PatchType, data []byte, subresources ...string) (result *v1alpha1.Cluster, err error)
+	ClusterExpansion
+}
+
+// clusters implements ClusterInterface
+type clusters struct {
+	client rest.Interface
+	ns     string
+}
+
+// newClusters returns a Clusters
+func newClusters(c *RookV1alpha1Client, namespace string) *clusters {
+	return &clusters{
+		client: c.RESTClient(),
+		ns:     namespace,
+	}
+}
+
+// Get takes name of the cluster, and returns the corresponding cluster object, and an error if there is any.
+func (c *clusters) Get(name string, options v1.GetOptions) (result *v1alpha1.Cluster, err error) {
+	result = &v1alpha1.Cluster{}
+	err = c.client.Get().
+		Namespace(c.ns).
+		Resource("clusters").
+		Name(name).
+		VersionedParams(&options, scheme.ParameterCodec).
+		Do().
+		Into(result)
+	return
+}
+
+// List takes label and field selectors, and returns the list of Clusters that match those selectors.
+func (c *clusters) List(opts v1.ListOptions) (result *v1alpha1.ClusterList, err error) {
+	result = &v1alpha1.ClusterList{}
+	err = c.client.Get().
+		Namespace(c.ns).
+		Resource("clusters").
```

Comment:
should this be defined in a const? For example `NamePlural`?

Reply:
this is generated code